### PR TITLE
[New Rule] Zoom Meeting with no Passcode

### LINKED ISSUE #281: [New Rule] Zoom Meeting With No Passcode
## Description
This rule identifies Zoom meetings that are created without a passcode. Meetings without a passcode are susceptible to
Zoombombing. Zoombombing is carried out by taking advantage of Zoom sessions that are not protected with a passcode.

Zoombombing refers to the unwanted, disruptive intrusion, generally by Internet trolls and hackers, into a video conference call. In a typical Zoombombing incident, a teleconferencing session is hijacked by the insertion of material that are lewd, obscene, racist, or antisemitic in nature, typically resulting of the shutdown of the session.

## Required Info

- **Eventing Sources:**
Filebeat with Zoom module enabled

- **Target Operating Systems:**
Windows, Linux, macOS

- **Platforms**
Zoom

- **Target ECS Version:** 1.6.0
- **New fields required in ECS for this?** NA
- **Related issues or PRs**

## Optional Info
- **References:**  
  * https://blog.zoom.us/a-message-to-our-users/
  * https://www.fbi.gov/contact-us/field-offices/boston/news/press-releases/fbi-warns-of-teleconferencing-and-online-classroom-hijacking-during-covid-19-pandemic

## Example Data
**Meeting with no passcode** (what the rule detects)
image
```
{
  "_index": "[redacted]",
  "_type": "_doc",
  "_id": "71SB7nMBKeQixZraigDT",
  "_version": 1,
  "_score": null,
  "_source": {
    "input": {
      "type": "http_endpoint"
    },
    "observer": {
      "product": "Webhook",
      "vendor": "Zoom"
    },
    "agent": {
      "name": "[redacted].local",
      "id": "[redacted]",
      "ephemeral_id": "[redacted]",
      "type": "filebeat",
      "version": "8.0.0"
    },
    "@timestamp": "2020-08-14T19:45:32.852Z",
    "ecs": {
      "version": "1.5.0"
    },
    "related": {
      "user": [
        "[redacted]",
        "[redacted]"
      ]
    },
    "service": {
      "type": "zoom"
    },
    "zoom": {
      "account_id": "[redacted]",
      "operator_id": "[redacted]",
      "meeting": {
        "start_time": "2020-08-14T19:45:00Z",
        "timezone": "America/Los_Angeles",
        "topic": "[redacted]",
        "id": [redacted],
        "type": 2,
        "uuid": "[redacted]",
        "host_id": "[redacted]"
      },
      "operator": "[redacted]"
    },
    "fileset": {
      "name": "webhook"
    },
    "event": {
      "duration": 3600000000000,
      "ingested": "2020-08-14T19:45:33.906313Z",
      "timezone": "-05:00",
      "kind": [
        "event"
      ],
      "module": "zoom",
      "action": "meeting.created",
      "type": [
        "info",
        "creation"
      ],
      "dataset": "zoom.webhook"
    },
    "tags": [
      "zoom-webhook",
      "forwarded"
    ]
  },
  "fields": {
    "event.ingested": [
      "2020-08-14T19:45:33.906Z"
    ],
    "@timestamp": [
      "2020-08-14T19:45:32.852Z"
    ],
    "zoom.meeting.start_time": [
      "2020-08-14T19:45:00.000Z"
    ],
    "suricata.eve.timestamp": [
      "2020-08-14T19:45:32.852Z"
    ]
  },
  "highlight": {
    "event.action": [
      "@kibana-highlighted-field@meeting.created@/kibana-highlighted-field@"
    ],
    "event.type": [
      "@kibana-highlighted-field@creation@/kibana-highlighted-field@"
    ],
    "event.kind": [
      "@kibana-highlighted-field@event@/kibana-highlighted-field@"
    ],
    "event.dataset": [
      "@kibana-highlighted-field@zoom.webhook@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1597434332852
  ]
}
```

### PR DESCRIPTION
**Pull request by peasead**
## Issues
Resolves #281 

## Summary
This rule identifies Zoom meetings that are created without a passcode. Meetings without a passcode are susceptible to
Zoombombing. Zoombombing is carried out by taking advantage of Zoom sessions that are not protected with a passcode.

Zoombombing refers to the unwanted, disruptive intrusion, generally by Internet trolls and hackers, into a video conference call. In a typical Zoombombing incident, a teleconferencing session is hijacked by the insertion of material that are lewd, obscene, racist, or antisemitic in nature, typically resulting of the shutdown of the session.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "https://www.fbi.gov/contact-us/field-offices/boston/news/press-releases/fbi-warns-of-teleconferencing-and-online-classroom-hijacking-during-covid-19-pandemic",
+]
+risk_score = 47
+rule_id = "58ac2aa5-6718-427c-a845-5f3ac5af00ba"
+severity = "medium"
+tags = ["Elastic", "SecOps", "Continuous Monitoring"]
+type = "query"
+
+query = '''
+event.kind:event and event.type:creation and event.module:zoom and event.dataset:zoom.webhook
```
Comment (brokensound77) [inline rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml:32]:
```suggestion
event.type:creation and event.module:zoom and event.dataset:zoom.webhook
```

**Review by threat-punter — APPROVED** [review]
🥃

**Comment by threat-punter** [comment]
I think the rule looks good, but I'm seeing this error when running the tests. 🤔

```
Loading rules from .../detection-rules/rules
Loaded 228 rules
Invalid rule file in .../detection-rules/rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml
Error at line:1,column:115
Unknown field
event.type:creation and event.module:zoom and event.dataset:zoom.webhook and event.action:meeting.created and not zoom.meeting.password:*
                                                                                                                  ^^^^^^^^^^^^^^^^^^^^^

Try adding event.module and event.dataset to specify beats module
```

**Comment by peasead** [comment]
> I think the rule looks good, but I'm seeing this error when running the tests. 🤔
> 
> 
> 
> ```
> 
> Loading rules from .../detection-rules/rules
> 
> Loaded 228 rules
> 
> Invalid rule file in .../detection-rules/rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml
> 
> Error at line:1,column:115
> 
> Unknown field
> 
> event.type:creation and event.module:zoom and event.dataset:zoom.webhook and event.action:meeting.created and not zoom.meeting.password:*
> 
>                                                                                                                   ^^^^^^^^^^^^^^^^^^^^^
> 
> 
> 
> Try adding event.module and event.dataset to specify beats module
> 
> ```

Yeah, this is the issue @brokensound77 mentioned that the field doesn't exist until the Zoom Filebeat module is merged to master.

File: rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+This rule identifies Zoom meetings that are created without a passcode. Meetings without a passcode are susceptible to
+Zoombombing. Zoombombing is carried out by taking advantage of Zoom sessions that are not protected with a passcode.
+Zoombombing refers to the unwanted, disruptive intrusion, generally by Internet trolls and hackers, into a video
+conference call. In a typical Zoombombing incident, a teleconferencing session is hijacked by the insertion of material
+that is lewd, obscene, racist, or antisemitic in nature, typically resulting of the shutdown of the session.
+"""
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Zoom Meeting with no Passcode"
```
Comment (bm11100) [inline rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml:19]:
```suggestion
name = "Zoom Meeting without Passcode"
```

File: rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+updated_date = "2020/09/15"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies Zoom meetings that are created without a passcode. Meetings without a passcode are susceptible to
+Zoombombing. Zoombombing is carried out by taking advantage of Zoom sessions that are not protected with a passcode.
+Zoombombing refers to the unwanted, disruptive intrusion, generally by Internet trolls and hackers, into a video
+conference call. In a typical Zoombombing incident, a teleconferencing session is hijacked by the insertion of material
+that is lewd, obscene, racist, or antisemitic in nature, typically resulting of the shutdown of the session.
```
Comment (bm11100) [inline rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml:14]:
```suggestion
Identifies Zoom meetings that are created without a passcode. Meetings without a passcode are susceptible to
Zoombombing. Zoombombing refers to the unwanted, disruptive intrusion, generally by Internet trolls and hackers, into a video
conference call. In a typical Zoombombing incident, a teleconferencing session is hijacked by the insertion of material
that is lewd, obscene, racist, or antisemitic in nature, typically resulting of the shutdown of the session.
```

Comment (bm11100) [reply rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml:14]: I think the below was repetitive, so suggested taking out the second sentence. You'll most likely need to re-lint if accepting the suggestion. 
> Meetings without a passcode are susceptible to Zoombombing. Zoombombing is carried out by taking advantage of Zoom sessions that are not protected with a passcode.

**Comment by peasead** [comment]
@brokensound77 @threat-punter here is the `fields.yml` file for the Zoom Filebeat module. Sharing because it shows the `meeting.password` field.

https://github.com/elastic/beats/blob/master/x-pack/filebeat/module/zoom/webhook/_meta/fields.yml

**Comment by brokensound77** [comment]
Once #304 merges, I am going to update this rule `metadata` with:
```toml
maturity = "development"
query_schema_validation = false
```

and approve it. Then, if the field makes it into the module, we can bump this to `production` within the release branch

File: rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+[metadata]
+creation_date = "2020/09/14"
+ecs_version = ["1.6.0"]
+maturity = "production"
```
Comment (brokensound77) [inline rules/cross-platform/initial-access-zoom-meeting-with-no-passcode.toml:4]:
```suggestion
maturity = "development"
query_schema_validation = false
```

Pending #304 merge

**Review by brokensound77 — APPROVED** [review]
Approving as development - will convert to production once the field is available and tested

**Pull request merged**